src/frontend/src/components/RecordSearch/Record/util.tsx

```diff
@@ -1,42 +1,69 @@
 import React from "react";
 import moment from "moment";
-import { ExpungementResultData } from "./types";
+import { ExpungementResultData, ChargeEligibilityStatus } from "./types";
 
 type EligibilityColor = "green" | "dark-blue" | "purple" | "red";
 
-// Status can come from either:
-// expungement_result.charge_eligibility.status or
-// summary.charges_grouped_by_eligibility_and_case keys
-export function getEligibilityColor(status: string, caseBalance = 0) {
+export function getEligibilityColor(
+  status: ChargeEligibilityStatus,
+  caseBalance = 0
+) {
   let color: EligibilityColor;
   let icon = "fa fa-";
 
   switch (status) {
-    case "Unknown":
-    case "Possible Eligible":
-    case "Possibly Will Be Eligible":
-    case "Needs More Analysis":
-      color = "purple";
-      icon += "question-circle";
+    case "Eligible Now":
+      color = caseBalance > 0 ? "dark-blue" : "green";
+      icon += caseBalance > 0 ? "dollar-sign" : "check";
       break;
     case "Ineligible":
       color = "red";
       icon += "circle-xmark";
       break;
-    case "Eligible Now":
-      color = caseBalance > 0 ? "dark-blue" : "green";
-      icon += caseBalance > 0 ? "dollar-sign" : "check";
-      break;
-    default:
+    case "Will Be Eligible":
       color = "dark-blue";
       icon += "clock";
+      break;
+    case "Possibly Eligible":
+    case "Possibly Will Be Eligible":
+    case "Unknown":
+    case "Needs More Analysis":
+    default:
+      color = "purple";
+      icon += "question-circle";
   }
 
   const bgColor = "bg-washed-" + (color === "dark-blue" ? "blue" : color);
 
   return { icon, color, bgColor };
 }
 
+function getShortLabel(
+  status: ChargeEligibilityStatus,
+  dateStr: string | null,
+  caseBalance: number
+) {
+  const date = moment(dateStr, "MMM/D/YY");
+
+  switch (status) {
+    case "Eligible Now":
+      if (caseBalance > 0) return "Eligible";
+      return status;
+    case "Will Be Eligible":
+      if (!dateStr || !date.isValid()) return status;
+      return "Eligible " + date.format("M/D/YY");
+    case "Needs More Analysis":
+      return "Needs Analysis";
+    case "Possibly Eligible":
+    case "Possibly Will Be Eligible":
+      return "Eligible?";
```

Comment:
Do you have a screenshot where this is relevant?

Reply:
The short labels are being used in the Summary and Counts sections of the expanded view:
<img width="508" alt="image" src="https://user-images.githubusercontent.com/120387970/220369629-b844d2c5-87a6-4e96-ae3a-dff5692c768f.png">
<img width="267" alt="image" src="https://user-images.githubusercontent.com/120387970/220369679-97956164-2043-4d14-bf66-6b9df8f3a94e.png">

In the Counts section, specific dates are removed and replaced with "Future". That was done in a subsequent commit.

The detailed display of the cases keeps the longer labels.

From what I could tell, the frontend charge eligibility statuses were mirroring the backend definitions.
https://github.com/codeforpdx/recordexpungPDX/blob/3852214c7eb5ca9460ac24a608ac76aa7e26085f/src/backend/expungeservice/models/expungement_result.py#L16-L23

Reply:
Nice!